client/src/components/overview/Overview.jsx

```diff
@@ -5,22 +5,240 @@ import AddToCart from './parts/addToCart/AddToCart.jsx';
+
+      this.setState({
+        thumbnailSection: 0
+      })
+    }
+
+  }
+
+  handleExpand() {
+    this.setState({
+      expanded: !this.state.expanded
+    })
+  }
+
+  // handles left or right main button click
+  mainSlide(dir) {
+
+    if (this.state.mainIndex === 0 && dir === -1) {
+      return
+    }
+
+    if (this.state.mainIndex === this.state.amount - 1 && dir > 0) {
+      return
+    }
+
+    if (this.state.mainIndex === 0) {
+      this.setState({
+        current: this.state.styles.results[0].photos[dir],
+        mainIndex: this.state.mainIndex + dir
+      })
+
+    } else {
+      this.setState({
+        current: this.state.styles.results[0].photos[this.state.mainIndex + dir],
+        mainIndex: this.state.mainIndex + dir
+      })
+    }
+
+  }
+
+  // handles thumbnail click
+  updateMainPic(index) {
+    this.setState({
+      mainIndex: index
+    })
+
+  }
+
+  getProductInfo(SKU) {
+
+  }
+
+  getGeneralInfo() {
+    const generalUrl = process.env.REACT_APP_API_OVERVIEW_URL + `products`;
+
+    fetch(generalUrl,
+      {
+        method: "GET",
+        headers:
+        {
+          "Content-Type": "application/json",
+          "Authorization": process.env.REACT_APP_API_OVERVIEW_TOKEN
+        }
+      }
+    )
+      .then(res => res.json())
+      .then((data) => {
+
+        // get general info (SKU and product info like name and description)
+        this.setState({
+          data: data,
+          SKU: data[0].id
+        })
+        // console.log(this.state);
+        return data;
+      })
+      .then((data) => {
+        const SKU = data[0].id;
+        const productUrl = process.env.REACT_APP_API_OVERVIEW_URL + `products/${SKU}/styles`;
+
+        fetch(productUrl,
+          {
+            method: "GET",
+            headers:
+            {
+              "Content-Type": "application/json",
+              "Authorization": process.env.REACT_APP_API_OVERVIEW_TOKEN
+            }
+          }
+        )
+        .then(res => res.json())
+
+        // get the info for the pictures, style and thumbnail
+
+        .then((data) => {
+
+          let thumbnails = data.results[0].photos;
+
+          let holder = [];
+          let box = [];
+
+          for (var i = 0; i < thumbnails.length; i++) {
+
+            thumbnails[i].index = i;
+            box.push(thumbnails[i]);
+
+            if (box.length === 7) {
+              holder.push(box);
+              box = [];
+            }
+
+            if (i >= thumbnails.length - 1) {
+              holder.push(box);
+              box = [];
+            }
+
+          }
+
+          this.setState({
+            styles: data,
+            current: data.results[0].photos[this.state.mainIndex],
+            amount: data.results[0].photos.length,
+            currentThumbnails: holder,
+            currentStyle: data.results[0]
+          })
+          // console.log('data from product', data);
+        })
+      })
+      .catch((err) => {
```

Comment:
This whole chain would be more readable with named functions imported from a controllers file. There's a lot of exposed complexity which makes it dense to read